tests/sundials/kinsol_08.cc

```diff
@@ -64,7 +64,7 @@ main()
 
   SUNDIALS::KINSOL<VectorType> kinsol(data);
 
-  kinsol.reinit_vector = [N](VectorType &v) { v.reinit(N); };
+  kinsol.reinit_vector = [N = N](VectorType &v) { v.reinit(N); };
```

Comment:
static const variables are by default captured. This may be a bug in the compiler that sees that the variable is `const` and initialized, and so promoted to `static`.

Reply:
see https://stackoverflow.com/questions/43467095/why-is-a-const-variable-sometimes-not-required-to-be-captured-in-a-lambda and https://godbolt.org/z/Ts9cr3EnE